merge.go

```diff
@@ -0,0 +1,659 @@
+	ret := make([]string, 0)
+	for _, x := range memLabels.Slice() {
+		if !dbLabels.Has(x) {
+			ret = append(ret, x)
+		}
+	}
+	if len(ret) == 0 {
+		return memLabels.Slice()
+	}
+	return ret
+}
+
+func setPropertiesDelta(delta delta, x withProperty, operationType operation) {
+	x.ForEachProperty(func(k string, v interface{}) bool {
+		pv, ok := v.(*ls.PropertyValue)
+		if !ok {
+			return true
+		}
+		v2, ok := x.GetProperty(k)
+		if !ok {
+			log.Printf("Error at %s: %v: Property does not exist", k, v)
+			return false
+		}
+		delta.setProp[k] = pv
+		pv2, ok := v2.(*ls.PropertyValue)
+		if !ok {
+			log.Printf("Error at %s: %v: Not property value", k, v)
+			return false
+		}
+		if !pv2.IsEqual(pv) {
+			log.Printf("Error at %s: Got %v, Expected %v: Values are not equal", k, pv, pv2)
+			return false
+		}
+		delta.setProp[k] = pv2
+		return true
+	})
+}
+
+func setNodeOverwrite(mem, db *lpg.Node, delta delta) {
+	for _, l := range db.GetLabels().Slice() {
+		if !mem.HasLabel(l) {
+			delta.removeLabel = append(delta.removeLabel, l)
+		}
+	}
+	db.ForEachProperty(func(s string, i interface{}) bool {
+		_, ok := mem.GetProperty(s)
+		if !ok {
+			delta.removeProp = append(delta.removeProp, s)
+		}
+		return true
+	})
+
+}
+func setEdgeOverwrite(mem, db *lpg.Edge, delta delta) {
+	if mem.GetLabel() != db.GetLabel() {
+		delta.removeLabel = append(delta.removeLabel, db.GetLabel())
+	}
+	db.ForEachProperty(func(s string, i interface{}) bool {
+		_, ok := mem.GetProperty(s)
+		if !ok {
+			delta.removeProp = append(delta.removeProp, s)
+		}
+		return true
+	})
+
+}
+
+// compares two lpg nodes and returns a diff
+func compareGraphNode(mem, db *lpg.Node, operationType operation) delta {
+	delta := delta{isNode: true, setProp: make(map[string]interface{})}
+	if db == nil {
+		delta.addLabel = mem.GetLabels().Slice()
+		delta.setProp = ls.CloneProperties(mem)
+		return delta
+	}
+	delta.addLabel = findNodeLabelDiff(mem.GetLabels(), db.GetLabels())
+	// Expected properties must be a subset
+	setPropertiesDelta(delta, mem, operationType)
+	if operationType == createOp {
+		setNodeOverwrite(mem, db, delta)
+	}
+	return delta
+}
+
+// compares two native lpg nodes and returns a diff
+func compareGraphEdge(mem, db *lpg.Edge, operationType operation) delta {
+	delta := delta{isNode: false, setProp: make(map[string]interface{})}
+	if db == nil {
+		delta.addLabel = []string{mem.GetLabel()}
+		delta.setProp = ls.CloneProperties(mem)
+		return delta
+	}
+	if mem.GetLabel() != db.GetLabel() {
+		delta.addLabel = []string{mem.GetLabel()}
+	}
+	setPropertiesDelta(delta, mem, updateOp)
+	if operationType == createOp {
+		setEdgeOverwrite(mem, db, delta)
+	}
+	return delta
+}
+
+type updateNode struct {
+	id     int64
+	labels []string
+	props  map[string]interface{}
+	n      *lpg.Node
+}
+
+type createNode struct {
+	id     int64
+	labels []string
+	props  map[string]interface{}
+	n      *lpg.Node
+}
+
```

Comment:
Write a func CreateNodes(Txn, []Delta, nodeIdMap)
This func should get all createNode deltas from the []Delta
Create all nodes, and update nodeIdMap